tests/acl/test_acl_l3v4v6.py

```diff
@@ -0,0 +1,1224 @@
+    pytest.mark.asic('marvell-teralynx'),
+    pytest.mark.disable_loganalyzer,  # Disable automatic loganalyzer, since we use it for the test
+    pytest.mark.topology("t0", "t1"),
+]
+
+MAX_WAIT_TIME_FOR_INTERFACES = 360
+
+BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+DUT_TMP_DIR = "acl_test_dir"  # Keep it under home dir so it persists through reboot
+FILES_DIR = os.path.join(BASE_DIR, "files")
+TEMPLATE_DIR = os.path.join(BASE_DIR, "templates")
+
+ACL_TABLE_TEMPLATE = "acltb_table.j2"
+ACL_REMOVE_RULES_FILE = "acl_rules_del.json"
+
+ACL_RULES_FULL_TEMPLATE = {
+    "ipv4": "acltbv4v6_v4_test_rules.j2",
+    "ipv6": "acltbv4v6_v6_test_rules.j2"
+}
+ACL_RULES_PART_TEMPLATES = {
+    "ipv4": tuple("acltbv4v6_test_rules_part_{}.j2".format(i) for i in range(1, 3)),
+    "ipv6": tuple("acltbv4v6_v6_test_rules_part_{}.j2".format(i) for i in range(1, 3))
+}
+
+DEFAULT_SRC_IP = {
+    "ipv4": "20.0.0.1",
+    "ipv6": "60c0:a800::5"
+}
+
+
+# TODO: These routes don't match the VLAN interface from the T0 topology.
+# This needs to be addressed before we can enable the v6 tests for T0
+DOWNSTREAM_DST_IP = {
+    "ipv4": "192.168.0.253",
+    "ipv6": "20c0:a800::14"
+}
+DOWNSTREAM_IP_TO_ALLOW = {
+    "ipv4": "192.168.0.252",
+    "ipv6": "20c0:a800::1"
+}
+DOWNSTREAM_IP_TO_BLOCK = {
+    "ipv4": "192.168.0.251",
+    "ipv6": "20c0:a800::9"
+}
+
+# Below M0_L3 IPs are announced to DUT by annouce_route.py, it point to neighbor mx
+DOWNSTREAM_DST_IP_M0_L3 = {
+    "ipv4": "192.168.1.65",
+    "ipv6": "20c0:a800:0:1::14"
+}
+DOWNSTREAM_IP_TO_ALLOW_M0_L3 = {
+    "ipv4": "192.168.1.66",
+    "ipv6": "20c0:a800:0:1::1"
+}
+DOWNSTREAM_IP_TO_BLOCK_M0_L3 = {
+    "ipv4": "192.168.1.67",
+    "ipv6": "20c0:a800:0:1::9"
+}
+
+# Below M0_VLAN IPs are ip in vlan range
+DOWNSTREAM_DST_IP_VLAN = {
+    "ipv4": "192.168.0.123",
+    "ipv6": "fc02:1000::5"
+}
+DOWNSTREAM_IP_TO_ALLOW_VLAN = {
+    "ipv4": "192.168.0.122",
+    "ipv6": "fc02:1000::6"
+}
+DOWNSTREAM_IP_TO_BLOCK_VLAN = {
+    "ipv4": "192.168.0.121",
+    "ipv6": "fc02:1000::7"
+}
+
+DOWNSTREAM_IP_PORT_MAP = {}
+
+UPSTREAM_DST_IP = {
+    "ipv4": "194.50.16.1",
+    "ipv6": "20c1:d180::11"
+}
+UPSTREAM_IP_TO_ALLOW = {
+    "ipv4": "193.191.32.1",
+    "ipv6": "20c1:cb50::12"
+}
+UPSTREAM_IP_TO_BLOCK = {
+    "ipv4": "193.221.112.1",
+    "ipv6": "20c1:e2f0::13"
+}
+
+VLAN_BASE_MAC_PATTERN = "72060001{:04}"
+
+LOG_EXPECT_ACL_TABLE_CREATE_RE = ".*Created ACL table.*"
+LOG_EXPECT_ACL_TABLE_REMOVE_RE = ".*Successfully deleted ACL table.*"
+LOG_EXPECT_ACL_RULE_CREATE_RE = ".*Successfully created ACL rule.*"
+LOG_EXPECT_ACL_RULE_REMOVE_RE = ".*Successfully deleted ACL rule.*"
+
+PACKETS_COUNT = "packets_count"
+BYTES_COUNT = "bytes_count"
+
+
+@pytest.fixture(scope="module", autouse=True)
+def remove_dataacl_table(duthosts):
+    """
+    Remove DATAACL to free TCAM resources.
+    The change is written to configdb as we don't want DATAACL recovered after reboot
+    """
+    TABLE_NAME = "DATAACL"
```

Comment:
Suggest checking if `DATAACL` is present before test and removing it if it's not. If `DATAACL` is not created by this test, it's better to leave it as is.